packages/eslint-plugin/docs/rules/i18n-text-domain.md

```diff
@@ -0,0 +1,27 @@
+# Enforce passing valid text domains (i18n-text-domain)
+
+[Translation functions](https://github.com/WordPress/gutenberg/blob/master/packages/i18n/README.md#api) must be called with a valid string literal as the text domain.
+
+## Rule details
+
+Examples of **incorrect** code for this rule:
+
+```js
+__( 'Hello World' ); // unless allowDefault is set.
+__( 'Hello World', 'default' ); // with allowDefault set.
+__( 'Hello World', foo );
+```
+
+Examples of **correct** code for this rule:
+
+```js
+__( 'Hello World' ); // with allowDefault set.
+__( 'Hello World', 'foo-bar' ); // with allowedTextDomains set
+```
+
+## Options
+
+This rule accepts two options:
+
+- Set the `allowDefault` boolean to allow omitting the text domain and using its default value (`'default'`).
+- Set `allowedTextDomains` to specify the list of allowed text domains, e.g. `'foo-bar'`.
```

Comment:
Should we configure this in Gutenberg to `'default'`, to enforce that _only_ the default text domain can be used?

I note that this would contradict what I've proposed in https://github.com/WordPress/gutenberg/pull/20555/files#r391095851 .

Reply:
Would you expect `allowedTextDomains = [ 'default' ]` to mean "only allow `default`, but remove it"? Because `default` is actually the default value. So `allowedTextDomains = [ 'default' ]` would basically mean `allowDefault = true`

Reply:
I don't really have strong opinions one way or the other, but yes, I think either of the following could be useful:

- Either eliminate `allowDefault`, and assume that the configuration is `allowedTextDomains: [ 'default' ]` would be the equivalent (including to encourage to remove it)
- Or... if `allowDefault` is kept, and is specified through a configuration, implicitly include this in `allowedTextDomains`

Reply:
WDYT about 6125f28f9b96c49ae0e374d0b743300cd16ef7e2?